realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1082,7 +1083,7 @@ public RealmAsyncTask executeTransaction(final Transaction transaction, final Tr
         // we can use to deliver the result
         if (callback != null && handler == null) {
             throw new IllegalStateException("Your Realm is opened from a thread without a Looper" +
-                    " and you provided a callback, we need a Handler to invoke your callback");
+                    " (or it has already been closed) and you provided a callback, we need a Handler to invoke your callback");
```

Comment:
Any reason or changing this when it is deprecated?
Also wouldn't it make more sense to check for `realm.isValid()` instead of doing this?